keanu-project/src/main/java/io/improbable/keanu/vertices/LogProbGraph.java

```diff
@@ -0,0 +1,35 @@
+package io.improbable.keanu.vertices;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+
+public class LogProbGraph {
+
+    private final Map<Vertex<?>, Vertex<?>> inputs;
```

Comment:
If I understand this correctly, there's no need to use `Vertex` as the map key. It would be enough to key on `VertexId` or even `VertexLabel`.

Reply:
Not all vertices have labels and if you key off of VertexId you will just end up calling `getInput(vertex.getId())` instead of `getInput(vertex)`.

Reply:
Yes but what I mean is, you could just as well use `String` for your key as `Vertex`.

Reply:
Then we have to have a consistent vertex to string mapping, which can be done (and I have on the tensorflow branch) but it's more complicated than we need at the moment.